idea/src/org/jetbrains/kotlin/idea/inspections/RemoveRedundantConstructorParenthesesInspection.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2000-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.idea.inspections
+
+import com.intellij.codeInspection.LocalQuickFix
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElementVisitor
+import org.jetbrains.kotlin.psi.*
+
+class RemoveRedundantConstructorParenthesesInspection : AbstractKotlinInspection() {
+
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+        return callExpressionVisitor { expression ->
+            val valueArguments = expression.valueArgumentList ?: return@callExpressionVisitor
+            if (valueArguments.arguments.size != 1) return@callExpressionVisitor
+            valueArguments.arguments.mapNotNull { it.getArgumentExpression() }.singleOrNull() as? KtLambdaExpression ?: return@callExpressionVisitor
+
+            holder.registerProblem(
+                    expression,
```

Comment:
I think we should report it on value argument and not on the whole call expression.